network-api/networkapi/wagtailpages/static/js/directory-listing-filters.js

```diff
@@ -0,0 +1,148 @@
+(function() {
+  // start of function
+
+  const profileCache = {};
+  const filters = document.querySelectorAll(`.profile-directory .fellowships-directory-filter .filter-option button`);
+  const profileContainer = document.querySelector(`.profiles .row`);
+  const { profileType, programType } = document.querySelector(`.profiles`).dataset;
+  const API_ENDPOINT = document.querySelector(`[data-api-endpoint]`).dataset.apiEndpoint;
+
+  /**
+   * After initial page load, the filter buttons are responsible for
+   * fetching results "per filter entry".
+   */
+  function getData(programYear) {
+    // set up a url for performing an API call:
+    let url = API_ENDPOINT;
+
+    // build and append the query arguments
+    let query = [
+      profileType ? `profile_type=${profileType}` : false,
+      programType ? `program_type=${programType}` : false,
+      programYear ? `program_year=${programYear}` : false
+    ].filter(v => v).join('&');
+
+    // and then perform our API call using the Fetch API
+    return new Promise(
+      (resolve, reject) => {
+        return fetch(`${url}&${query}`)
+               // conver the resonse from JSON to real data
+               .then(response => response.json())
+               // all went well? resolve.
+               .then(obj => resolve(obj))
+               // errors? reject.
+               .catch(error => reject(error));
+      }
+    );
+  };
+
+  /**
+   * Load an array of profiles into the profile car container.
+   * We first map the profiles array, turning profile objects
+   * into templated HTML using the same HTML as we have in
+   * profile_blocks.html
+   */
+  function loadResults(profiles) {
+    let cards = profiles.map(profile => {
```

Comment:
Note that this block (L46 to L88) is effectively "blocked by whatever is the final form implemented in the fellow-style PR", since it needs to be the same HTLM code as used in the loop in `profile_blocks.html`